index.js

```diff
@@ -555,9 +557,93 @@ const plugin = fp(async function (app, opts) {
       operationName
     })
 
+    /* istanbul ignore next */
+    if (execution.initialResult) {
+      const acceptHeader = reply.request.raw.headers.accept
+
+      if (
+        !(
+          acceptHeader &&
+          new Negotiator({
+            headers: { accept: acceptHeader }
+          }).mediaType([
+            // mediaType() will return the first one that matches, so if the client
+            // doesn't include the deferSpec parameter it will match this one here,
+            // which isn't good enough.
+            MEDIA_TYPES.MULTIPART_MIXED_NO_DEFER_SPEC,
+            MEDIA_TYPES.MULTIPART_MIXED_EXPERIMENTAL
+          ]) === MEDIA_TYPES.MULTIPART_MIXED_EXPERIMENTAL
+        )
+      ) {
+        // The client ran an operation that would yield multiple parts, but didn't
+        // specify `accept: multipart/mixed`. We return an error.
+        throw new Error(
```

Comment:
Please create an error in https://github.com/mercurius-js/mercurius/blob/master/lib/errors.js and use it here